packages/@webex/plugin-meetings/src/multistream/remoteMediaGroup.ts

```diff
@@ -0,0 +1,222 @@
+  preferLiveVideo?: boolean; // applies only to groups of type MC.MediaType.VideoMain and MC.MediaType.VideoSlides
+};
+
+export class RemoteMediaGroup {
+  private mediaRequestManager: MediaRequestManager;
+
+  private priority: number;
+
+  private options: Options;
+
+  private remoteMedia: RemoteMedia[];
+
+  private mediaRequestId?: MediaRequestId; // id of the "active-speaker" media request id
+
+  private pinnedRemoteMedia: RemoteMedia[];
+
+  constructor(
+    mediaRequestManager: MediaRequestManager,
+    receiveSlots: ReceiveSlot[],
+    priority: number,
+    commitMediaRequest: boolean,
+    options: Options = {}
+  ) {
+    this.mediaRequestManager = mediaRequestManager;
+    this.priority = priority;
+    this.options = options;
+
+    this.remoteMedia = receiveSlots.map(
+      (slot) =>
+        new RemoteMedia(slot, this.mediaRequestManager, {
+          resolution: this.options.resolution,
+        })
+    );
+    this.pinnedRemoteMedia = [];
+
+    this.sendActiveSpeakerMediaRequest(commitMediaRequest);
+  }
+
+  /**
+   * Gets the array of all remote media elements from the group (excluding the pinned ones)
+   *
+   * @returns {Array<RemoteMedia>}
+   */
+  public getRemoteMedia() {
+    // return a shallow copy so that the client cannot modify this.remoteMedia array
+    return [...this.remoteMedia];
+  }
+
+  /**
+   * Gets the array of all pinned remote media elements from the group
+   *
+   * @returns {Array<RemoteMedia>}
+   */
+  public getPinnedRemoteMedia() {
+    // return a shallow copy so that the client cannot modify this.pinnedRemoteMedia array
+    return [...this.pinnedRemoteMedia];
+  }
+
+  /**
+   * Pins a specific remote media instance to a specfic CSI, so the media will
+   * no longer come from active speaker, but from that CSI.
+   * If no CSI is given, the current CSI value is used.
+   *
+   */
+  public pin(remoteMedia: RemoteMedia, csi?: CSI): void {
+    // if csi is not specified, use the current one
+    const targetCsi = csi || remoteMedia.csi;
+
+    if (!targetCsi) {
+      throw new Error(
+        `failed to pin a remote media object ${remoteMedia.id}, because it has no CSI set and no CSI value was given`
+      );
+    }
+
+    if (this.pinnedRemoteMedia.indexOf(remoteMedia) >= 0) {
+      if (targetCsi === remoteMedia.csi) {
+        // remote media already pinned to target CSI, nothing to do
+        LoggerProxy.logger.log(
+          `RemoteMediaGroup#pin --> remote media ${remoteMedia.id} already pinned`
+        );
+
+        return;
+      }
+    } else {
+      const idx = this.remoteMedia.indexOf(remoteMedia);
+
+      if (idx < 0) {
+        throw new Error(
+          `failed to pin a remote media object ${remoteMedia.id}, because it is not found in this remote media group`
+        );
+      }
+
+      this.remoteMedia.splice(idx, 1);
+      this.pinnedRemoteMedia.push(remoteMedia);
+
+      this.cancelActiveSpeakerMediaRequest(false);
```

Comment:
unnecessary since it's done in sendActiveSpeakerMediaRequest

Reply:
yeah but I prefer to make it more explicit